.github/workflows/build.yaml

```diff
@@ -4,27 +4,46 @@ on:
   push:
     branches: [ rolling ]
   workflow_dispatch:
+  schedule:
+    # Run every morning to detect flakiness and broken dependencies
+    - cron: '03 5 * * *'
 defaults:
   run:
     shell: bash
 jobs:
-  test:
-    runs-on: ubuntu-latest
+  build_and_test:
+    name: build_and_test_${{ matrix.BUILD_TYPE }}_${{ matrix.ROS_DISTRO }}
     strategy:
       fail-fast: false
       matrix:
-        distro: ['iron', 'jazzy', 'rolling']
+        include:
+          # Rolling (source)
+          - ROS_DISTRO: rolling
+            BUILD_TYPE: source
+            DOCKER_IMAGE: ubuntu:noble
+          # Jazzy (binary)
+          - ROS_DISTRO: jazzy
+            BUILD_TYPE: binary
+            DOCKER_IMAGE: ubuntu:noble
```

Comment:
I think we should remove this `DOCKER_IMAGE` flag as it's one extra item to configure.
See commend below for relevant suggestion.